packages/rfq-indexer/api/src/queries/proofsQueries.ts

```diff
@@ -1,15 +1,25 @@
 import { db } from '../db'
 
 // typical fields to return for a BridgeProofProvided event when it is joined to a BridgeRequest
-export const qProofs = () => {
-  return db
+export const qProofs = ({ activeOnly}: { activeOnly: boolean} = { activeOnly: false}) => {
+  let query = db
     .selectFrom('BridgeProofProvidedEvents')
+    .leftJoin('BridgeProofDisputedEvents', (join) =>
+      join
+        .onRef('BridgeProofDisputedEvents.transactionId', '=', 'BridgeProofProvidedEvents.transactionId')
+        .onRef('BridgeProofDisputedEvents.blockTimestamp', '>', 'BridgeProofProvidedEvents.blockTimestamp')
+    )
     .select([
       'BridgeProofProvidedEvents.transactionId as transactionId_proof',
       'BridgeProofProvidedEvents.blockNumber as blockNumber_proof',
       'BridgeProofProvidedEvents.blockTimestamp as blockTimestamp_proof',
       'BridgeProofProvidedEvents.transactionHash as transactionHash_proof',
-
       'BridgeProofProvidedEvents.relayer as relayer_proof',
-    ])
-}
+    ]);
+
+  if (activeOnly) {
```

Comment:
If there is a proof, and ensuing dispute, and then a new proof, woudl the new proof appear here? given that we are filtering for transactionId of the relayed transaction, wouldnt the new (valid) proof not appear?

Could be wrong here

Reply:
Nevermind, the leftjoin solves for this